File: codegen/results/v002/qwen3.5_27b-nvfp4_nothink/20260412_032821/python/003_fibonacci/solution.py
import sys

def main():
    # Check if a command line argument is provided
    if len(sys.argv) != 2:
        return

    try:
        n = int(sys.argv[1])
    except ValueError:
        return

    if n < 1:
        return

    a, b = 1, 1

    # Print the first number if it doesn't exceed N
    while a <= n:
        print(a)
        # Move to the next Fibonacci number
        a, b = b, a + b

if __name__ == "__main__":
    main()

File: codegen/results/v002/qwen3.5_27b-nvfp4_nothink/20260412_032821/typescript/003_fibonacci/solution.ts
// Parse the command-line argument
const n = parseInt(process.argv[2], 10);

// If N is less than 1 or invalid, print nothing and exit
if (isNaN(n) || n < 1) {
  process.exit(0);
}

// Initialize the first two Fibonacci numbers in the sequence: 1, 1, ...
let a = 1;
let b = 1;

// Print the first number if it is <= N
if (a <= n) {
  console.log(a);
}

// Generate and print subsequent numbers while they are <= N
while (b <= n) {
  console.log(b);
  // Calculate the next Fibonacci number
  const next = a + b;
  a = b;
  b = next;
}